### PR TITLE
Project Euler Problem 80

### PR DESCRIPTION
**Pull request by sandeepgupta007**
### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: project_euler/problem_80/sol1.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+"""
+    Euler Problem : 80
```
Comment (dhruvmanila) [inline project_euler/problem_80/sol1.py:2]: - Please remove the indentation, it's not needed for module level docstring
- Simplify by removing @ symbol

File: project_euler/problem_80/sol1.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    answer = 0
+    dot100 = decimal.Context(prec=105)
+    for i in range(2, 100):
+        number = decimal.Decimal(i)
+        sqrt_number = number.sqrt(dot100)
+        if len(str(sqrt_number)) > 1:
+            answer += int(str(sqrt_number)[0])
+            sqrt_number = str(sqrt_number)[2:101]
+            answer += sum([int(x) for x in sqrt_number])
+        return answer
```
Comment (dhruvmanila) [inline project_euler/problem_80/sol1.py:24]: I think your `return` statement is at wrong indentation level which is resulting in the wrong answer: https://travis-ci.com/github/TheAlgorithms/Python/jobs/395341755#L318-L321

File: project_euler/problem_80/sol1.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    a few extra places for decimal otherwise there will be rounding
+    error.
+    @answer     : 40886
+"""
+import decimal
+
+
+def solution() -> int:
+    answer = 0
+    dot100 = decimal.Context(prec=105)
```
Comment (dhruvmanila) [inline project_euler/problem_80/sol1.py:16]: Is it possible for a better variable name than `dot100` ?

File: project_euler/problem_80/sol1.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    @Solution: Used decimal python module to calculate the decimal
+    places up to 100, the most important thing would be take calculate
+    a few extra places for decimal otherwise there will be rounding
+    error.
+    @answer     : 40886
+"""
+import decimal
+
+
+def solution() -> int:
```
Comment (dhruvmanila) [inline project_euler/problem_80/sol1.py:14]: Please include `doctest` and function `docstring` explaining in brief what this function does.

**Review by dhruvmanila — CHANGES_REQUESTED** [review]

File: project_euler/problem_80/sol1.py (hunk 4 above)
Comment (sandeepgupta007) [reply project_euler/problem_80/sol1.py:12]: Apologies, updated as per all the comments.

**Review by dhruvmanila — APPROVED** [review]

**Pull request merged**